src/pages/index.tsx

```diff
@@ -24,37 +20,55 @@ export default function Home({
     wantToRead,
     previously_read,
   };
-  
   return (
-    <div className="bg-slate-100 text-center">
-      <div className="bg-slate-100 text-start ml-56">
-        <WelcomeMessage name={"Yazan"} />
-        <div className="flex">
-          <div className="w-1/4">
+    <div className="bg-slate-100 min-h-screen py-8 px-4">
+      <div className="max-w-screen-xl mx-auto space-y-20">
+        <WelcomeMessage />
+        
+        <div className="flex flex-col md:flex-row gap-12">
+          <div className="md:w-1/4">
             <CurrentlyReading books={currentlyReading} />
           </div>
-
-          <div className="w-3/4 ">
+          <div className="md:w-3/4">
             <WantToRead books={wantToRead} />
           </div>
         </div>
+
         <BooksSection books={allBooks} />
-        
       </div>
     </div>
   );
 }
 export async function getServerSideProps() {
   const res = await fetch(
-    "https://bookclub-backend.nn.r.appspot.com/api/v1/98aac522330f4c29882dcfd3736822ad/shelves"
+    `${process.env.PUBLIC_API}/98aac522330f4c29882dcfd3736822ad/shelves`
   );
   const data = await res.json();
+  const currentlyReading = data.currently_reading.books.map(mapBookData);
```

Comment:
This is a good practice - you should always map the data once fetched - so components don't have to worry about it